plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -340,7 +340,7 @@ The default behaviour is to detect which access style to use based on the config
             storageClass, pathStyleAccess);
 
         AmazonS3 client = s3Service.client(metadata.settings(), maxRetries, useThrottleRetries, pathStyleAccess);
-        blobStore = new S3BlobStore(settings, client, bucket, serverSideEncryption, bufferSize, maxRetries, cannedACL, storageClass);
```

Comment:
@rjernst this is a good idea, which I took a little bit further and refactored in f2ef1985b75d381612e3af985dff906d1922c049.  Let me know what you think.